backend/api/MQTT/MqttService.cs

```diff
@@ -0,0 +1,186 @@
+                    break;
+                case Type type when type == typeof(IsarTask):
+                    OnIsarTopicReceived<IsarTask>(content);
+                    break;
+                case Type type when type == typeof(IsarStep):
+                    OnIsarTopicReceived<IsarStep>(content);
+                    break;
+                default:
+                    _logger.LogWarning(
+                        "No callback defined for MQTT message type '{type}'",
+                        messageType.Name
+                    );
+                    break;
+            }
+        }
+
+        private void OnConnected(MqttClientConnectedEventArgs obj)
+        {
+            _logger.LogInformation("Successfully connected.");
+        }
+
+        private void OnConnectingFailed(ManagedProcessFailedEventArgs obj)
+        {
+            _logger.LogWarning(
+                "Couldn't connect to MQTT broker. Exception: {message}\n"
+                    + "      Retrying in {time}s",
+                obj.Exception.Message,
+                _reconnectDelay.Seconds
+            );
+        }
+
+        private void OnDisconnected(MqttClientDisconnectedEventArgs obj)
+        {
+            _logger.LogInformation("Successfully disconnected.");
+        }
+
+        private void RegisterCallbacks()
+        {
+            _mqttClient.ConnectedHandler = new MqttClientConnectedHandlerDelegate(OnConnected);
+            _mqttClient.DisconnectedHandler = new MqttClientDisconnectedHandlerDelegate(
+                OnDisconnected
+            );
+            _mqttClient.ConnectingFailedHandler = new ConnectingFailedHandlerDelegate(
+                OnConnectingFailed
+            );
+            _mqttClient.ApplicationMessageReceivedHandler =
+                new MqttApplicationMessageReceivedHandlerDelegate(OnMessageReceived);
+        }
+
+        public void SubscribeToTopics(List<string> topics)
+        {
+            List<MqttTopicFilter> topicFilters = new();
+            topics.ForEach(topic => topicFilters.Add(new MqttTopicFilter() { Topic = topic }));
+            _mqttClient.SubscribeAsync(topicFilters);
+        }
+
+        private void OnIsarTopicReceived<T>(string content) where T : MqttMessage
+        {
+            var message = JsonSerializer.Deserialize<T>(content);
+            if (message is null)
+            {
+                _logger.LogError(
+                    "Could not create '{className}' object from MQTT message json",
+                    nameof(T)
+                );
+                return;
+            }
+
+            var type = typeof(T);
+            try
+            {
+                var raiseEvent = type switch
+                {
+                    _ when type == typeof(IsarMission) => MqttIsarMissionReceived,
+                    _ when type == typeof(IsarTask) => MqttIsarTaskReceived,
+                    _ when type == typeof(IsarStep) => MqttIsarStepReceived,
+                    _
```

Comment:
Should be possible to use the same style for a switch on type as mentioned in the previous comment?

Reply:
It's not possible in either, because we are not checking the **type** of the variable, we are checking the **value** of the variable, which is a type.